tests/epics/demo/test_demo.py

```diff
@@ -199,6 +199,28 @@ async def test_retrieve_mock_and_assert(mock_mover: demo.Mover):
     )
 
 
+async def test_mocks_in_device_share_parent():
+    mock = Mock()
+    async with DeviceCollector(mock=mock):
+        mock_mover = demo.Mover("BLxxI-MO-TABLE-01:Y:")
+
+    assert get_mock(mock_mover) is mock
+    assert get_mock(mock_mover.setpoint) is mock.setpoint
+    assert get_mock_put(mock_mover.setpoint) is mock.setpoint.put
+    await mock_mover.setpoint.set(10)
+    get_mock_put(mock_mover.setpoint).assert_called_once_with(10, wait=ANY)
+
+    await mock_mover.velocity.set(100)
+    await mock_mover.setpoint.set(67)
+
+    mock.assert_has_calls(
+        [
+            call.velocity.put(100, wait=True),
+            call.setpoint.put(67, wait=True),
+        ]
+    )
```

Comment:
would prefer to be explicit about the previous calls (assert_has_calls does a subset check):
```suggestion
    mock.reset_mock()
    await mock_mover.velocity.set(100)
    await mock_mover.setpoint.set(67)

    assert mock.call_args_list == [
        call.velocity.put(100, wait=True),
        call.setpoint.put(67, wait=True),
    ]
```